sonar-application/src/test/java/org/sonar/application/ClusterTest.java

```diff
@@ -0,0 +1,163 @@
+import java.net.SocketException;
+import java.util.Enumeration;
+import java.util.Properties;
+import org.junit.Assume;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.sonar.process.Props;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ClusterTest {
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void test_cluster() throws Exception {
+    ClusterProperties clusterProperties = initDefaultProperties();
+
+    try (Cluster cluster = new Cluster(clusterProperties)) {
+      assertThat(
+        cluster.isActive()
+      ).isEqualTo(true);
+    }
+
+    clusterProperties.setEnabled(false);
+    try (Cluster cluster = new Cluster(clusterProperties)){
+      assertThat(
+        cluster.isActive()
+      ).isEqualTo(false);
+    }
+  }
+
+  @Test
+  public void test_interface() throws Exception {
+    ClusterProperties clusterProperties = initDefaultProperties();
+    String ipAddress = findIPv4Address().getHostAddress();
+
+    if (ipAddress != null) {
+      int port = findAvailablePort();
+      clusterProperties.setInterfaces(ipAddress);
+      clusterProperties.setPort(port);
+
+      try (Cluster cluster = new Cluster(clusterProperties)){
+        assertThat(
+          cluster.hazelcastInstance.getConfig().getNetworkConfig().getInterfaces().isEnabled()
+        ).isEqualTo(true);
+        assertThat(
+          cluster.hazelcastInstance.getConfig().getNetworkConfig().getInterfaces().getInterfaces()
+        ).containsExactly(ipAddress);
+        assertThat(
+          ((InetSocketAddress) cluster.hazelcastInstance.getLocalEndpoint().getSocketAddress()).getPort()
+        ).isEqualTo(port);
+        assertThat(
+          ((InetSocketAddress) cluster.hazelcastInstance.getLocalEndpoint().getSocketAddress()).getAddress().getHostAddress()
+        ).isEqualTo(ipAddress);
+      }
+    }
+  }
+
+  @Test
+  public void test_with_already_used_port() throws IOException {
+    ClusterProperties clusterProperties = initDefaultProperties();
+    InetAddress ipAddress = findIPv4Address();
+
+    Assume.assumeTrue(ipAddress != null);
+    ServerSocket socket = null;
+    Cluster cluster = null;
+    try {
+      socket = new ServerSocket(0, 100, ipAddress);
+      clusterProperties.setInterfaces(ipAddress.getHostAddress());
+      clusterProperties.setPort(socket.getLocalPort());
+      expectedException.expect(HazelcastException.class);
+      cluster = new Cluster(clusterProperties);
+    } finally {
+      if (socket != null) {
+        socket.close();
+      }
+      if (cluster != null) {
+        cluster.hazelcastInstance.getLifecycleService().terminate();
+      }
+    }
+  }
+
+  @Test
+  public void test_adding_port_to_members() throws Exception {
+    ClusterProperties clusterProperties = initDefaultProperties();
+    InetAddress ipAddress = findIPv4Address();
+
+    Assume.assumeTrue(ipAddress != null);
```

Comment:
given that `Assume.assumeTrue` raises an exception, I think you would move it to method `findIPv4Address` and save on some duplication